### PR TITLE
Add multi-platform build job

### PR DESCRIPTION
**Pull request by niconosenzo**
This change adds a new build job that allows building a multi-platform image.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -64,3 +70,115 @@ jobs:
...
+          aws-secret-access-key: ${{ secrets.ORG_AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Login to Amazon ECR
+        uses: aws-actions/amazon-ecr-login@v2
+      
+      - name: Create manifest list and push
+        working-directory: /tmp/digests
+        run: |
+          docker buildx imagetools create $(jq -cr '.tags | map("-t " + .) | join(" ")' <<< "$DOCKER_METADATA_OUTPUT_JSON") \
```
Comment (alexeyklyukin) [inline .github/workflows/build.yaml:179]: How is this DOCKER_METADATA_OUTPUT_JSON defined?

File: .github/workflows/build.yaml [hunk 2]
```diff
@@ -64,3 +70,115 @@ jobs:
...
+        with:
+          aws-access-key-id: ${{ secrets.ORG_AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.ORG_AWS_SECRET_ACCESS_KEY }}
+          aws-region: us-east-1
+
+      - name: Login to Amazon ECR
+        uses: aws-actions/amazon-ecr-login@v2
+      
+      - name: Create manifest list and push
+        working-directory: /tmp/digests
```
Comment (alexeyklyukin) [inline .github/workflows/build.yaml:177]: How do we use tose digests expect for setting a work directory there?

**Review by alexeyklyukin — APPROVED** [review]
Left some questions, but overall looks solid (unauthoritative review, I don't have much expertice in gh actions).

File: .github/workflows/build.yaml (hunk 1 above)
Comment (niconosenzo) [reply .github/workflows/build.yaml:179]: It's created [above](https://github.com/timescale/cloud-actions/pull/40/files#diff-d0777657fa3fd81d23aaf7273e58aee453b04e67882517900c56daeef9b3e4c1R96-R102) by the [metadata action](https://github.com/docker/metadata-action), it exports some information as environment variables.

File: .github/workflows/build.yaml (hunk 2 above)
Comment (niconosenzo) [reply .github/workflows/build.yaml:177]: Not sure if I understood the question, but in any case, digests are downloaded and stored into `/tmp/digests` a few steps above, and used here by buildx to create the multiplatform image. [This a similar example of use](https://docs.docker.com/build/ci/github-actions/multi-platform/#distribute-build-across-multiple-runners) of this pattern.

File: .github/workflows/build.yaml (hunk 1 above)
Comment (alexeyklyukin) [reply .github/workflows/build.yaml:179]: Thank you for the explanation!

File: .github/workflows/build.yaml (hunk 2 above)
Comment (alexeyklyukin) [reply .github/workflows/build.yaml:177]: Sorry, too many typos. I was wondering how the digests being downloaded are being used, you've answered the right question.Seems the 
```
$(printf '${{ env.REGISTRY_IMAGE }}@sha256:%s ' *)
```
actually goes through all file names in /tmp/digest and produce image references.

Thank you for the explanation!

File: .github/workflows/build.yaml [hunk 3]
```diff
@@ -64,3 +70,115 @@ jobs:
...
+          - platform: arm64
+            runs_on: ARM64
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        
+      - name: Prepare
+        run: |
+          platform=linux/${{ matrix.platform }}
+          echo "PLATFORM_PAIR=${platform//\//-}" >> $GITHUB_ENV
```
Comment (graveland) [inline .github/workflows/build.yaml:94]: platform contains 2 parts: linux and arm64, but PLATFORM_PAIR contains only one- kinda confusing?

File: .github/workflows/build.yaml [hunk 4]
```diff
@@ -64,3 +70,115 @@ jobs:
...
+    runs-on: [ "self-hosted", "${{ matrix.runs_on }}" ]
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [ "amd64", "arm64" ]
+        include:
+          - platform: amd64
+            runs_on: dev-us-east-1
+          - platform: arm64
+            runs_on: ARM64
```
Comment (graveland) [inline .github/workflows/build.yaml:86]: nit: pretty sure runs_on is case-insensitive, and ARM64 looks to be yelling to me :)

**Review by graveland — APPROVED** [review]
lgtm!

This is great btw- I'd love to get multiplatform images for deployer/operator/popper/snapper for my dev environment work, and this should make it a lot easier!

File: .github/workflows/build.yaml (hunk 3 above)
Comment (niconosenzo) [reply .github/workflows/build.yaml:94]: Actually both are pairs:
```bash
echo "PLATFORM_PAIR=${platform//\//-}"
PLATFORM_PAIR=linux-amd64
```
Anyway I will make the naming more clear.

Comment (graveland) [reply .github/workflows/build.yaml]: Ha! oops- looking closer helps, sorry about the noise.

**Pull request merged**